packages/cli/src/commands/site/deploy/index.ts

```diff
@@ -14,54 +14,118 @@
-import type { SiteDeployArgs } from "./SiteDeployArgs.js";
+import type { SiteDeployArgs } from "./siteDeployArgs.js";
+function deployHandler(
+  siteConfig?: SiteConfig,
+): yargs.CommandModule<CommonSiteArgs, SiteDeployArgs> {
+  const directory = siteConfig?.directory;
+  const autoVersion = siteConfig?.autoVersion;
+  const gitTagPrefix = autoVersion?.tagPrefix;
 
-export const command: CommandModule<
-  CommonSiteArgs,
-  SiteDeployArgs
-> = {
-  command: "deploy",
-  describe: "Deploy an uploaded version",
-  builder: (argv) => {
-    return argv
-      .options({
-        "siteVersion": {
-          type: "string",
-          conflicts: "clearVersion",
-          // group: "Deploy Version",
-          // implies: { "clearVersion": "false" },
-        },
-        undeploy: {
-          alias: "clearVersion",
-          description: "Causes the site to no longer be deployed",
-          type: "boolean",
-          conflicts: "siteVersion",
-          // group: "Deploy Version",
-          // implies: { "siteVersion": "" },
-        },
-      }).group(
-        ["siteVersion", "clearVersion"],
-        "Version To Deploy (requires one of)",
-      );
+  const command: yargs.CommandModule<
+    CommonSiteArgs,
+    SiteDeployArgs
+  > = {
+    command: "deploy",
+    describe: "Deploy an uploaded version",
+    builder: (argv) => {
+      return argv
+        .options({
+          directory: {
+            type: "string",
+            description: "Directory to deploy",
+            ...directory
+              ? {
+                default: directory,
+              }
+              : {
+                demandOption: true,
+              },
+          },
+          uploadOnly: {
+            type: "boolean",
+            description: "Upload the directory but do not deploy it",
+            default: false,
+          },
+          version: {
+            type: "string",
+            description: "Version to deploy",
+            ...autoVersion == null
+              ? { conflicts: "autoVersion" }
+              : {},
+          },
+          autoVersion: {
+            type: "string",
+            description:
+              "Enables autoversioning. Can be set to 'git-describe' to use git describe to determine the version.",
+            ...(autoVersion != null)
+              ? { default: autoVersion.type }
+              : { conflicts: "version" },
+          },
+          gitTagPrefix: {
+            type: "string",
+            description:
+              "Prefix to match git tags against when --autoVersion=git-describe is used. If not provided, a default prefix 'v' is used.",
+            ...gitTagPrefix
+              ? {
+                default: gitTagPrefix,
+              }
+              : {},
+          },
+        }).group(
+          ["autoVersion", "gitTagPrefix"],
+          "Autoversion Arguments",
+        )
+        .group(
+          ["version", "directory", "uploadOnly"],
+          "Common Arguments",
```

Comment:
Are you mentioning that it would be better to move these arguments on each of the subcommands? So that when you write a non-complete command like `osdk unstable site version` you won't get these "common" arguments filed?